independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/gradle-kotlin-dsl/kotlin/build.tpl.qute.gradle.kts

```diff
@@ -15,10 +15,10 @@ allOpen {
 }
 
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
-    {#if java.version == "11"}
-    kotlinOptions.jvmTarget = JavaVersion.VERSION_11.toString()
+    {#if java.version == "17"}
+    kotlinOptions.jvmTarget = JavaVersion.VERSION_17.toString()
     {#else}
-    kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()
+    kotlinOptions.jvmTarget = JavaVersion.VERSION_11.toString()
```

Comment:
My understanding is that the Java version should be resolved properly before arriving here. And given they now have all the same format, maybe we could just change things to:
```
kotlinOptions.jvmTarget = JavaVersion.VERSION_{java.version}.toString()
```
?
And same for the other one?

Reply:
I thought about it, but if the user has java version which is incompatible with kotlin such as java 18, should we fallback to 11, or put 18 ?

Reply:
If it's using the same logic used for the other codestarts, there is a mechanism to choose the best LTS version and it's always a LTS version that is injected for Maven for instance so I think it should probably be good enough for Gradle/Kotlin.
Now, there could be a moment where the last LTS is not supported by Kotlin but in this case, we should probably have a mechanism where I adjusted this: https://github.com/quarkusio/quarkus/pull/24645/files#diff-e3a379680d4a534804bfaf1f30197ae16808192379a05287be973b77585a8cc6R26 .
We could have a constant just with the others with the max LTS version supported by Kotlin.

Reply:
@gsmet I pushed a fix